cmd/buildah/config.go

```diff
@@ -135,13 +135,15 @@ func updateConfig(builder *buildah.Builder, c *cli.Context) {
 		}
 	}
 	if c.IsSet("entrypoint") {
-		entrypointSpec, err := shellwords.Parse(c.String("entrypoint"))
-		if err != nil {
-			logrus.Errorf("error parsing --entrypoint %q: %v", c.String("entrypoint"), err)
-		} else {
-			builder.SetEntrypoint(entrypointSpec)
-			builder.SetCmd(nil)
+		entrypointSpec := make([]string, 3)
+		entrypointSpec[0] = "/bin/sh"
+		entrypointSpec[1] = "-c"
+		entrypointSpec[2] = c.String("entrypoint")
```

Comment:
I am curious why we are not validating `c.String("entrypoint")` as we used to before.

In addition, what happens if someone puts in `entrypoint` such a string `"ls -lha /tmp"`. Can you add a test for such a case?

Reply:
@ripculd I'm not 100% sure what you mean by "not validating".  I think you mean not calling shellwords.Parse() on the string any more.  The reason for that change is illustrated nicely by your example.    With the current code, if you were to inspect the container after setting the entrypoint as you suggested you'll see:

```
            "Entrypoint": [
                "/bin/sh",
                "-c",
                "ls -lha /tmp"
```
which matches Docker and the result if you use ENTRYPOINT in the Dockerfile passed to `buildah bud`.  Prior to this change when we called shellwords.Parse, the results of `buildah inspect` would be:

```
            "Entrypoint": [
                "/bin/sh",
                "-c",
                "ls",
                "-lha", 
                "/tmp"
```
Which did not match Docker and with some entrypoints such as "ping www.google.com" would cause an unexpected result or failure.  

I'll see if I can work something up for testing.

```
#######
Test:
# buildah config --entrypoint "ls -lha /tmp" $cid
# buildah run $cid
total 0
drwxrwxrwt    2 root     root           6 Jan  9 19:37 .
drwxr-xr-x    1 root     root          28 Apr 23 13:31 ..
```

Reply:
Got it :bulb: ! Thanks for the clarification.